<launch>
  <arg name="filter_unknown" default="false" />
  <arg name="min_x" default="-300.0" />
  <arg name="max_x" default="300.0" />
  <arg name="min_y" default="-300.0" />
  <arg name="max_y" default="300.0" />
  <arg name="min_z" default="-300.0" />
  <arg name="max_z" default="300.0" />

  <arg name="input_topic" default="/input/tracked_objects" />
  <arg name="inside_output_topic" default="/output/inside_tracked_objects" />
  <arg name="outside_output_topic" default="/output/outside_tracked_objects" />

  <node
    pkg="tracked_objects_crop_box_filter"
    exec="tracked_objects_crop_box_filter_node"
    name="$(anon tracked_objects_crop_box_filter_node)"
    output="screen">

    <!-- Remap topics if needed -->
    <remap from="input/objects" to="$(var input_topic)" />
    <remap from="output/inside_objects" to="$(var inside_output_topic)" />
    <remap from="output/outside_objects" to="$(var outside_output_topic)" />

    <!-- Parameters -->
    <param name="filter_unknown" value="$(var filter_unknown)" />
    <param name="min_x" value="$(var min_x)" />
    <param name="max_x" value="$(var max_x)" />
    <param name="min_y" value="$(var min_y)" />
    <param name="max_y" value="$(var max_y)" />
    <param name="min_z" value="$(var min_z)" />
    <param name="max_z" value="$(var max_z)" />
  </node>
</launch>
